### PR TITLE
privacy-center: Introduce fides-consent script package 

### LINKED ISSUE #1395: Expose Fides Consent Cookie for Browser Integrations - Client side Javascript
We want to reflect different data uses explicitly and expose them for customers and users to quickly integrate with, in plain English.

As a marketer or web developer, I want to expose the Fides consent cookie in such a way that I can easily interact with it via code directly on my website or with tag managers made by Google, Adobe, etc.

**Details**

The sample consent configuration listed below is part of our privacy center config. We want to avoid the arbitrary grouping of cookies and instead provide a reasonable set of default mappings from consent to cookie, so that we support standard use cases out of the box. The mappings will also allow users to create their own cookies should they want to name things differently.

**AC** 
- Given that a user includes a Fides javascript widget in their `<head>` when the page loads, then the consent object is set based on the cookie
- Given that a fides cookie is set based on the cookie config of the privacy center, when a user includes our fides javascript widget, then the properties of the cookie are made available on the window.
- Given that a user updates their cookie, when a page reloads with the fides script, then the fides consent object is updated to reflect the cookie.
- Given that a user has no Fides cookie on their browser, when a page with the Fides script reloads, then the consent object is set to the default configurations of the privacy center. - Note - this requires the script to read from the privacy center config to get the defaults. Can this be done?

``` 
{
    "title": "Take control of your data",
    "description": "When you use our services, you’re trusting us with your information. We understand this is a big responsibility and work hard to protect your information and put you in control.",
    "serverurldevelopment": "http://localhost:8080/api/v1",
    "serverurlproduction": "http://localhost:8080/api/v1",
    "logopath": "/logo.svg",
    "actions": [
        {
            "policykey": "examplerequestpolicy",
            "iconpath": "/download.svg",
            "title": "Access your data",
            "description": "We will email you a report of the data related to your account.",
            "identityinputs": {
                "name": "optional",
                "email": "required",
                "phone": "optional"
            }
        },
        {
            "policykey": "exampleerasurepolicy",
            "iconpath": "/delete.svg",
            "title": "Erase your data",
            "description": "We will delete all of your account data. This action cannot be undone.",
            "identityinputs": {
                "name": "optional",
                "email": "required",
                "phone": "optional"
            }
        }
    ],
    "includeConsent": true,
    "consent": {
        "cookieName": "ethycaconsent",
        "consentOptions": [
            {
                "fidesDataUseKey": "thirdpartysharing",
                "name": "Share my information",
                "description": "...",
                "highlight": true,
                "url": "https://example.com/privacy#data-sales",
                "cookieKeys": [
                    "datasharing",
                    "googleanalytics"
                ],
            },
            {
                "default": true,
                "fidesDataUseKey": "advertising",
                "name": "Sell my information",
                "url": "https://example.com/privacy#provide-service",
                "highlight": false,
                "description": "Manage how we use your data, including Do Not Sell My Personal Information.",
                "cookieKeys": [
                    "fb",
                    "gads"
                ]
            },
            {
                "default": true,
                "fidesDataUseKey": "improve",
                "name": "Provide a service",
                "url": "https://example.com/privacy#provide-service",
                "highlight": false,
                "description": "Manage how we use your data, including Do Not Sell My Personal Information.",
                "cookieKeys": [
                    "google_analytics",
                    "mixpanel"
                ],
            },
            {
                "default": true,
                "fidesDataUseKey": "provide.service",
                "name": "Provide a service",
                "url": "https://example.com/privacy#provide-service",
                "highlight": false,
                "description": "Manage how we use your data, including Do Not Sell My Personal Information."
            }
        ]
    }
}
```

### PR DESCRIPTION
**Pull request by ssangervasi**
Closes #1395

### Code Changes

* Adds `fides-consent.js`, exposing the our cookie parsing logic in a browser script.
* E2E tests the consent set->get flow using a standalone page.
* The privacy center uses the same code from the module for its cookie logic.

### Steps to Confirm

* [ ] Build and test the script
    * `npm install` and `npm run cy:start` should put everything in place automatically
* [ ] Try it in dev
    * The demo page should reflect the changes made in the privacy center.
    *  http://localhost:3000/fides-consent-demo.html

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation Updated:
  * [x] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  * [ ] documentation issue created (tag docs-team to complete issue separately)
* [x] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

Actually writing the script was easy, but figuring out a good build + test flow was quite interesting. I used rollup, just like we have in fidesui. Next.js wouldn't let me use `<script>` tag with a static asset, so for testing I created a static HTML page that is simply served as-is. This is probably better for isolation anyway.

### CODE REVIEW

**Review by LKCSmith — APPROVED** [review]
I think the code looks good, but I get a 404 page doesn't exist when trying to access `http://localhost:3000/fides-consent-demo.html` -- does a flag need to be active?

**Comment by ssangervasi** [comment]
> I think the code looks good, but I get a 404 page doesn't exist when trying to access `http://localhost:3000/fides-consent-demo.html` -- does a flag need to be active?

@LKCSmith - Running `npm run dev` from the `privacy-center` directory works for me. And `npm run cy:start` hosts a built version.
